src/args.ts

```diff
@@ -10,17 +11,30 @@ export interface ToolchainOptions {
     components: string[] | undefined,
 }
 
-export function toolchain_args(): ToolchainOptions {
+export function toolchain_args(overrideFile: string): ToolchainOptions {
     let components: string[] | undefined = input.getInputList('components');
     if (components && components.length === 0) {
         components = undefined;
     }
+
     return {
-        name: input.getInput('toolchain', {required: true}),
+        name: determineToolchain(overrideFile),
         target: input.getInput('target') || undefined,
         default: input.getInputBool('default'),
         override: input.getInputBool('override'),
         profile: input.getInput('profile') || undefined,
         components: components,
     };
 }
+
+function determineToolchain(overrideFile: string): string {
+    if (existsSync(overrideFile)) {
+        info(`Using toolchain override from ${overrideFile}`);
+        return readFileSync(overrideFile, {
+            encoding: "utf-8",
+            flag: "r"
+        })
```

Comment:
Can you add `String.trim()` here? I'm not sure how exactly `rustup` invocation later will behave if one of arguments will contain a line break or other whitespace character.

Reply:
Ah good catch! I was actually thinking of that but then somehow forgot about it. Will add some tests for this as-well :)

Reply:
Ah good catch! I was actually thinking of that but then somehow forgot about it. Will add some tests for this as-well :)

Reply:
Please see 0cea29e :)